### ISSUE TITLE
Starscream dependency is outdated (conflicts with Apollo)

### ISSUE THREAD

**Issue by callumboddy** [aws-amplify/amplify-ios#1290]
**Describe the bug**
Starscream dependency is outdated

**To Reproduce**
1. Install Apollo via SPM @ latest 0.44.0 (>= 0.41.0)
2. Install Amplify via SPM @ latest 1.10.0

**Expected behavior**
Resolves packages successfully

**Environment(please complete the following information):**
 - Amplify Framework Version: 1.10.0
 - Amplify Categories: All
 - Dependency Manager: SwiftPM
 - Swift Version: 5.4
 - CLI Version: 5.0.1
 - Xcode Version: 12.5

**Additional context**
```
Dependencies could not be resolved because root depends on 'apollo-ios' 0.44.0..<1.0.0 and root depends on 'amplify-ios' 1.10.0..<2.0.0.
'amplify-ios' is incompatible with 'apollo-ios' because 'apollo-ios' >= 0.42.0 depends on 'Starscream' 4.0.4..<4.1.0.
'amplify-ios' practically depends on 'Starscream' 3.1.1..<3.2.0 because 'amplify-ios' depends on 'aws-appsync-realtime-client-ios' 1.4.3..<2.0.0 and 'aws-appsync-realtime-client-ios' depends on 'Starscream' 3.1.1..<3.2.0.
```

**Comment by callumboddy** [comment]
This can be resolved by downgrading to Apollo client 0.40.0 but it looks like they are using later version, however having looked at the latest releases of Starscream, they are not marked as `verified`.

**Comment by diegocstn** [comment]
hey @callumboddy, 
we've considered updating Starscream dependency to v4.0.4 but it's a non-trivial task as the new version has a few breaking changes and the APIs are drastically different.
